lib/cdo/poste.rb

```diff
@@ -85,8 +85,13 @@ def self.unsubscribe(email, hashed_email, params={})
         unsubscribed_ip: params[:ip_address],
       )
     else
+      is_dashboard_user = DASHBOARD_DB[:users].where(hashed_email: hashed_email).any?
+      is_dashboard_student = DASHBOARD_DB[:users].
+        where(hashed_email: hashed_email).
+        first[:user_type] == 'student' if is_dashboard_user
```

Comment:
same suggestion as above, and since this is used twice maybe consolidate into a function in `poste.rb`?